handlers/me.go

```diff
@@ -0,0 +1,39 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/speedrun-website/leaderboard-backend/database"
+	"github.com/speedrun-website/leaderboard-backend/graph/model"
+	"github.com/speedrun-website/leaderboard-backend/middleware"
+)
+
+func MeHandler(c *gin.Context) {
+	user, _ := c.Get(middleware.JwtConfig.IdentityKey)
+	db, err := database.GetDatabase()
+
+	// todo error handler or middleware?
+	if err != nil {
+		log.Println("Unable to connect to database", err)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	var me model.User
+	result := db.Where(model.User{
+		Email: user.(*model.User).Email,
+	}).First(&me)
+
+	if result.Error != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
```

Comment:
Should this be a 500? Doesn't this mean the user wasn't found for whatever reason? If I'm understanding this right it should probably be a 401.

Reply:
I'm not entirely sure how to handle this, does that looks good?

```go
user, _ := c.Get(middleware.JwtConfig.IdentityKey)
db, err := database.GetDatabase()

// todo error handler or middleware?
if err != nil {
	log.Println("Unable to connect to database", err)
	c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
		"message": err.Error(),
	})
	return
}

var me model.User
result := db.Where(model.User{
	Email: user.(*model.User).Email + "lmao",
}).First(&me)

if result.Error != nil {
	var code = http.StatusInternalServerError

	if errors.Is(result.Error, gorm.ErrRecordNotFound) {
		code = http.StatusNotFound
	}

	c.AbortWithStatusJSON(code, gin.H{
		"message": result.Error.Error(),
	})
	return
}

c.JSON(http.StatusOK, me)
```

Using `errors.Is` is what was recommanded by [gorm's documentation](https://gorm.io/docs/error_handling.html). Thoughts ?

Reply:
Sorry I missed this comment. This pattern seems sound to me, but I think the code should be 401 (Unauthorized) rather than 404 (Not Found). I think there's debate for either side though, I'm probably fine with either.